src/main/java/org/springframework/retry/annotation/AnnotationAwareRetryOperationsInterceptor.java

```diff
@@ -476,14 +477,16 @@ private boolean isTemplate(String expression) {
 				&& expression.contains(PARSER_CONTEXT.getExpressionSuffix());
 	}
 
-	private <T> T evaluate(Expression expression, Class<T> type) {
-		RetryContext context = RetrySynchronizationManager.getContext();
+	private <T> T evaluate(Expression expression, Class<T> type, boolean stateless) {
 		Args args = null;
-		if (context != null) {
-			args = (Args) context.getAttribute("ARGS");
-		}
-		if (args == null) {
-			args = Args.NO_ARGS;
+		if (stateless) {
+			RetryContext context = RetrySynchronizationManager.getContext();
+			if (context != null) {
+				args = (Args) context.getAttribute("ARGS");
+			}
+			if (args == null) {
+				args = Args.NO_ARGS;
+			}
 		}
 		return expression.getValue(this.evaluationContext, args, type);
```

Comment:
Is it expected to propagate an `args`  var as `null` for statefull unlike for stateless with an `Args.NO_ARGS`?

Reply:
Yes; I want them to get a hard error `EL1007E: Property or field 'args' cannot be found on null`.

I failed to save the doc change for this; pushing now...